From git@z Thu Jan  1 00:00:00 1970
Subject: [PATCH] cifs: fix enum usage
From: trix@redhat.com <trix@redhat.com>
Date: Mon, 28 Mar 2022 06:03:00 -0700
Message-Id: <20220328130300.3090213-1-trix@redhat.com>
To: sfrench@samba.org, nathan@kernel.org, ndesaulniers@google.com
Cc: linux-cifs@vger.kernel.org, samba-technical@lists.samba.org, linux-kernel@vger.kernel.org, llvm@lists.linux.dev, Tom Rix <trix@redhat.com>
List-Id: <llvm.lists.linux.dev>
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 7bit

From: Tom Rix <trix@redhat.com>

Clang build fails with
cifsfs.c:709:18: error: implicit conversion from enumeration
  type 'enum statusEnum' to different enumeration
  type 'enum tid_status_enum'
  tcon->status = CifsExiting;
               ~ ^~~~~~~~~~~

The type of the element status changed, so enum needed to change.
Replace CifsExiting with TID_EXITING.

Fixes: 7e5c8c02911b ("smb3: cleanup and clarify status of tree connections")
Signed-off-by: Tom Rix <trix@redhat.com>
Reviewed-by: Nathan Chancellor <nathan@kernel.org>
Link: https://lore.kernel.org/r/20220328130300.3090213-1-trix@redhat.com
---
 fs/cifs/cifsfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/cifs/cifsfs.c b/fs/cifs/cifsfs.c
index 0789bf1496c74..a47fa44b6d52b 100644
--- a/fs/cifs/cifsfs.c
+++ b/fs/cifs/cifsfs.c
@@ -706,7 +706,7 @@ static void cifs_umount_begin(struct super_block *sb)
 		spin_unlock(&cifs_tcp_ses_lock);
 		return;
 	} else if (tcon->tc_count == 1)
-		tcon->status = CifsExiting;
+		tcon->status = TID_EXITING;
 	spin_unlock(&cifs_tcp_ses_lock);
 
 	/* cancel_brl_requests(tcon); */ /* BB mark all brl mids as exiting */

-- 
2.26.3


